src/Result.ts

```diff
@@ -1,83 +1,85 @@
-export interface Result<T, U> {
-	isOk(): boolean;
-	isErr(): boolean;
-	mapOk<R>(mapper: (t: T) => R): Result<R, U>;
-	mapErr<R>(mapper: (u: U) => R): Result<T, R>;
-	unwrapOrThrow(errToThrow?: Error): T;
-	unwrapErrOrThrow(): U;
-}
-
-class OkResult<T, U> implements Result<T, U> {
+class OkResult<T> {
 	private readonly contents: T;
 
 	constructor(contents: T) {
 		this.contents = contents;
 	}
 
-	isOk() {
+	isOk(): this is OkResult<T> {
 		return true;
 	}
 
-	isErr() {
+	isErr(): false {
 		return false;
 	}
 
-	mapOk<R>(mapper: (t: T) => R): Result<R, U> {
+	mapOk<R>(mapper: (t: T) => R): OkResult<R> {
 		return new OkResult(mapper(this.contents));
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	mapErr<R>(mapper: (u: U) => R): Result<T, R> {
-		return this as unknown as Result<T, R>;
+	mapErr(): OkResult<T> {
```

Comment:
Is the return type intentional? I'm not familiar with JS or TS internals so I can't comment much.